src/k8s-engine/unixlikeIntegrations.ts

```diff
@@ -0,0 +1,136 @@
+const console = new Console(Logging.background.stream);
+const PUBLIC_LINK_DIR = '/usr/local/bin';
+
+/*
+ * There are probably going to be only two kinds of integrations: WSL for Windows,
+ * and everything else for macos and linux, should it be supported. For now, this class
+ * is currently intended to be used for various VM strategies for macOS.
+ */
+export default class UnixlikeIntegrations {
+  #results: Record<string, boolean | string> = {}
+  /*
+   * Used to supply integration-warnings
+   */
+  protected pathConflictManager = new PathConflictManager();
+
+  /*
+   * We want to watch ${PUBLIC_LINK_DIR} for changes, but fs.watch won't watch it until it exists.
+   * Once there's a watcher on it, the directory can be deleted, and when it's recreated the
+   * original watcher will still work on it.
+   */
+  #binWatcher: fs.FSWatcher|null = null;
+
+  constructor() {
+    this.setupBinWatcher();
+  }
+
+  async listIntegrations(): Promise<Record<string, boolean | string>> {
+    for (const name of INTEGRATIONS) {
+      const linkPath = path.join(PUBLIC_LINK_DIR, name);
+      const desiredPath = resources.executable(name);
+
+      try {
+        const currentDest = await fs.promises.readlink(linkPath);
+
+        if (currentDest === desiredPath) {
+          this.#results[linkPath] = true;
+        } else {
+          this.#results[linkPath] = `Already linked to ${ currentDest }`;
+        }
+      } catch (error) {
+        if (error.code === 'ENOENT') {
+          this.#results[linkPath] = false;
+        } else if (error.code === 'EINVAL') {
+          this.#results[linkPath] = `File exists and is not a symbolic link`;
+        } else {
+          this.#results[linkPath] = `Can't link to ${ linkPath }: ${ error }`;
+        }
+      }
+    }
+
+    return this.#results;
+  }
+
+  async setIntegration(linkPath: string, state: boolean): Promise<string | undefined> {
+    const desiredPath = resources.executable(path.basename(linkPath));
+
+    this.#results[linkPath] = '';
+    if (state) {
+      try {
+        await fs.promises.symlink(desiredPath, linkPath);
+      } catch (err) {
+        const message = `Error creating symlink for ${ linkPath }:`;
+
+        console.error(message, err);
+        this.#results[linkPath] = `${ message } ${ err.message }`;
+
+        return this.#results[linkPath] as string;
+      }
+    } else {
+      try {
+        await fs.promises.unlink(linkPath);
+      } catch (err) {
+        const message = `Error unlinking symlink for ${ linkPath }`;
+
+        console.error(message, err);
+        this.#results[linkPath] = `${ message } ${ err.message }`;
+
+        return this.#results[linkPath] as string;
+      }
+    }
+  }
+
+  listIntegrationWarnings(): void {
+    for (const name of INTEGRATIONS) {
+      this.pathConflictManager.reportConflicts(name);
+    }
+    window.send('k8s-integration-warnings', 'docker', ["Links to rancher-desktop's nerdctl"]);
```

Comment:
That clobbers the warning from `reportConflicts` (kind of, depends on races, because we're not waiting for the promise).

Reply:
Sorry, I didn't see these comments before the merge. As you saw, I found the problem here once everything merged.